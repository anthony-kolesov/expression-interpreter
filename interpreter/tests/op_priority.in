# Confirm that operations execute in proper priority

out 5 + 4 * 3
print "\n"

out 4 * 3 + 5
print "\n"

out 5 - 4 * 3
print "\n"

out 4 * 3 - 5
print "\n"

out 5 + 12 / 3
print "\n"

out 12 / 3 + 5
print "\n"

out 80 - 21 / 7
print "\n"

out 21 / 7 - 80
print "\n"

out 4 ^ 3 + 5
print "\n"

out 5 + 4 ^ 3
print "\n"

out 4 ^ 3 - 5
print "\n"

out 5 - 4 ^ 3
print "\n"

out 4 ^ 3 * 2
print "\n"

out 2 * 4 ^ 3
print "\n"

out 4 ^ 3 / 2
print "\n"

out 256 / 4 ^ 3
print "\n"
