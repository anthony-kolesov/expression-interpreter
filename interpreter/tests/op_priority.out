17
17
-7
7
9
9
77
-77
69
69
59
-59
128
128
32
4
